components/mpas-albany-landice/src/mode_forward/mpas_li_advection.F

```diff
@@ -947,6 +964,13 @@ subroutine apply_mass_balance(&
       sfcMassBalApplied(:) = sfcMassBal(:)
       basalMassBalApplied(:) = basalMassBal(:)
 
+      ! Initialize bare ice ablation and bare ice ablation applied fields 
+      bareIceAblation(:) = 0.0_RKIND 
+      where (sfcMassBal < 0.0_RKIND)
+            bareIceAblation = -1.0_RKIND * sfcMassBal
+      end where
+      bareIceAblationApplied(:) = bareIceAblation(:)
```

Comment:
`bareIceAblationApplied` should be calculated from `sfcMassBalApplied` which will catch the situation where there is insufficient ice in a grid cell to satisfy the ablation demand of `bareIceAblation`

Reply:
You are suggesting that 'bareIceAblationApplied' should be calculated as:
`bareIceAblationApplied = -1.0_RKIND * sfcMassBalApplied` 
?

Reply:
On this note, do we need `bareIceAblation` at all? It seems like only the `Applied` field will ever be used.

Reply:
I had also been thinking we would use the difference between bareIceAblation and bareIceAblationApplied to check for conservation issues.  But maybe we should be doing that with sfcMassBal /  sfcMassBalApplied so as to be evaluating the full field from ELM, not just the ablation portion.

Regardless of that choice, bareIceAblationApplied should be coming from sfcMassBalApplied (as opposed to sfcMassBal), since that is what MALI actually used in its mass balance.

Reply:
Here's what bareIceAblation (left) and bareIceAblationApplied (right) look like zoomed in on one particular area of the margin:

![Screenshot 2025-04-08 at 4 57 09 PM](https://github.com/user-attachments/assets/b19f9faa-e767-486a-82b4-c1a006dce0d8)

So we are probably violating energy conservation to some extent here because the full ablation field is more extensive than the applied one (I've confirmed that the footprint shape for bareIceAblationApplied maps to the ice thickness footprint).

Reply:
Interesting.  Do these differences represent changes in ice extent from one coupling interval to another?  I.e. are we seeing a lag in the ice mask?  I can't figure out why it would look so similar but not identical.

Reply:
I'm not sure. I can try to check that though as I have two times steps / coupling intervals worth of data in the same file I used to make this figure.

Reply:
Resolving this for now as we are removing the bareIceAblation field (in favor of using the sfcMassBal and sfcMassBalApplied fields for tracking this problem, if it exists).